src/state_machine.rs

```diff
@@ -143,6 +143,27 @@ impl State {
         }
     }
 
+    pub fn matches_our_prefix(&self, name: &XorName) -> Result<bool, RoutingError> {
+        match *self {
+            State::Elder(ref state) => Ok(state.our_prefix().matches(name)),
+            State::Adult(ref state) => Ok(state.our_prefix().matches(name)),
+            State::BootstrappingPeer(_) | State::JoiningPeer(_) | State::Terminated => {
+                Err((InterfaceError::InvalidState).into())
+            }
+        }
+    }
+
+    pub fn closest_elders_to<'a>(
+        &'a self,
+        name: &XorName,
```

Comment:
XorName is copy. so it is ok to pass by value.
and if you do, I think you can drop explicit lifetimes.

Not sure it is better?

Reply:
`XorName` has nothing to do with the lifetimes here. It's the return type that must be valid for the duration of `self` as it's a ref from something somewhere within `self`.

Reply:
Ok, I thought lifetime to self were implicit if no other arguments were a reference, but maybe not in this context.